test/common/TestCommon.hs

```diff
@@ -0,0 +1,369 @@
+{-# LANGUAGE DataKinds, RankNTypes, TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+module TestCommon (dReverse0, dReverse1, fX, fX1Y, fXXY, fXYplusZ, fXtoY, freluX,
```

Comment:
fX, fX1Y, fXXY, fXYplusZ, fXtoY, freluX and probably a lot of others are not really auxiliary, but one-time silly tests. However, no harm factoring them out, using some of them in more than one test (but, really, most of them are silly and trivial), making sure nothing is repeated, inlining again everything that is only used in one place.

Reply:
I was expecting that.